base/sparse/sparsematrix.jl

```diff
@@ -93,11 +91,15 @@ function Base.showarray(io::IO, S::SparseMatrixCSC;
     pad = ndigits(max(S.m,S.n))
     k = 0
```

Comment:
I've been thinking about this `k` too. It is quite strange, the loop makes k go over a range but then k is modified in the loop.

This way of printing also makes printing matrices with very large dimensions slow. 

```jl
julia> a = spzeros(10^8, 10^8);

julia> tic(); display(a); toc()
100000000×100000000 sparse matrix with 0 Float64 nonzero entries:
elapsed time: 0.25458145 seconds
0.25458145
```